### PR TITLE
create a contributing.md

### PR DESCRIPTION
**Pull request by martinagvilas**
Reference to issue #63 

This is just a copy of @dujm file

### CODE REVIEW

**Comment by martinagvilas** [comment]
This is just a copy of @dujm file

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # pandas-mentoring
 <img src="img/logo.jpeg" width="600" height="200"><br>
-
+
```
Comment (datapythonista) [inline README.md]: Do you want to revert this?

Surely not a problem at this point, but in projects like pandas, unrelated changes should be avoid (even if they aren't wrong, they change the git history, and we prefer to keep things under control by being strict at those things).

**Review by datapythonista — COMMENTED** [review]
lgtm, just one small comment

@dujm do you want to have a look at this?

**Review by dujm — APPROVED** [review]
Thanks for leading this @martinagvilas! The ```contributing.md``` looks good! @datapythonista  

Now we can add more best practices in the ```md```. Btw, here is one reference we could use [pandas-mentoring/doc/source/development/contributing.rst ](https://github.com/python-sprints/pandas-mentoring/blob/master/doc/source/development/contributing.rst)

File: README.md (hunk 1 above)
Comment (martinagvilas) [reply README.md]: Sure! Do you have a suggestion on how to revert this?

Comment (datapythonista) [reply README.md]: the "correct" way is probably some sort of `git reset`, but I think it's better in this case to simply edit the file, remove the spaces in that line, and add a new commit to this PR

**Comment by datapythonista** [comment]
> Now we can add more best practices in the `md`. Btw, here is one reference we could use [pandas-mentoring/doc/source/development/contributing.rst](https://github.com/python-sprints/pandas-mentoring/blob/master/doc/source/development/contributing.rst)

We already have that file copied to this repo. :)

We'll see what we can do, but eventually we may want to merge the guide you started with the pandas contributing documentation, so anyone who wants to contribute to pandas can benefit from it. But this is a lot of work, will take time until we see exactly what we want to do. :)

**Review by datapythonista — CHANGES_REQUESTED** [review]
I think something didn't work as expected. ;)

One tip. Before pushing to a PR already open (or before opening it to), you can run:
```
git checkout <your-pr-branch>
git fetch upstream
git diff upstream/master
```

This will show you the same as https://github.com/python-sprints/pandas-mentoring/pull/63/files, but before pushing. If you see that there are unrelated changes, then you probably won't want to push.

**Comment by martinagvilas** [comment]
I'm having an issue that I don't fully understand. My branch master now has the correct stucture that I want for my pull request, but my branch contributing-guide does not. When I try to merge my branch master with by branch contributing-guide, the merge does not happen but it does not throw any issues. Should I close this PR and try again creating a new branch from my master branch and then creating a new pull request?

**Comment by dujm** [comment]
@martinagvilas  I don't know what ```branch contributing-guide``` is.  But I think it's really nice that you want to understand the problem so that we can all learn from it. So how about we just leave this PR open? Maybe others can answer your question and then you could try with this branch again. 

Meanwhile, exactly as you said, a faster way is to create a new branch and send a new PR (only pickup the commit id for adding ```contribution.md```)

```
  # 1) Add a new branch from your origin/master branch
   git branch new-branch origin/master
   git checkout new-branch
   
   # 2) Make a commit of your work (e.g. contribution.md)
   git add contribution.md
   git commit -m 'add contribution.md'

   # 3) If you had multiple commits, but you want to select a specific commit(e.g. commit id: bbb909f) for a pull request 
   git cherry-pick bbb909f
   
   # 4) Push to your new-branch
   git push origin new-branch 
```

**Comment by datapythonista** [comment]
@martinagvilas I'd say the problem is that you're merging from your outdated master. Let me explain how I see the workflow to update a branch.

Git is distributed, and from git prespective [all branches are created equal](http://www.ushistory.org/Declaration/document/), and any branch of any fork is as good as the rest. For convenience and by convention, we all consider the `master` branch of the https://github.com/python-sprints/pandas-mentoring the "good" one. When a commit reaches that branch, becomes part of our official code, and all the rest of branches and forks are somehow temporary copies to work on.

So, what we want is to get what's in that specific branch, into the branch where you are working. The first step is to download it locally:
- `git fetch upstream`

This will create an `upstream/master` branch, that is read-only, and an exact copy of the "good" branch, but locally.

Then we want to be in the branch to update, to make it active, if it's not we do:
- `git checkout contributing-guide`

And finally, we tell git to copy all the commits (the ones that are not already there), from the branch `upstream/master` to `contributing-guide` with:
- `git merge upstream/master`

Note that in the whole process I never used or changed your local `master` (which is different from `upstream/master`). I have the feeling that you're doing the process correctly, except that instead of merging from `upstream/master`, you're merging from `master`. Quite reasonable, and those git things are far from intuitive at the beginning, but since your `master` is probably behind `contributing-guide`, weird things are happening.

Does this make sense?

**Comment by martinagvilas** [comment]
Hi Marc, makes perfect sense! Thank you for the explanation. 

However, I tried what you said, my master is now updated with upstream/master, but when I try to merge upstream/master inside my contributing-guide branch, git says "Already up to date". But when I inspect the contents of this branch it only shows three files (the readme, the contributing and the learning points markdowns, thus missing the folders and the .gitignore), which I find super weird and do not want to push the changes as I will break this PR even more. Do you have an idea of why this might be and how to solve it?

**Comment by martinagvilas** [comment]
I think I got it now. Apparently, I was stuck in a rebase. Could you check if now is alright? Sorry for the troubles and thanks for the explanation!

**Comment by datapythonista** [comment]
There are still files being deleted, but looks better. If merging says you're already up to date, it means that you have a commit in your branch deleting those files. The merge only adds new commits from the "good" branch, but if you have a commit from for example fixing a conflict that didn't go well and deleted those file, this won't be fixed by the merge.

One solution could be check in the commits of this branch which is the one deleting the file and revert it (`git revert <commit-id>` if I'm not wrong).

Another one that I use somehow often is in your branch do a `git reset --soft upstream/master`. This will undo all the commits, and leave all the changes in the staging area. Then with `git reset` you can unstage the changes you don't want to commit, and just leave the ones that you want. After the commit when you push, first try it normally, but it will fail and then you'll have to push with the `-f` flag. This is because you rewrote the history and git won't let you overwrite the history in a push unless you're explicit you want it.

**Comment by sparalic** [comment]
@martinagvilas I started sifting throw all of the emails to capture some of the feedback @datapythonista and other shared when folks got stuck. I also took a look at the contribution file when I was uploading it initially and I agree with @dujm. We should leverage as much of that work as possible. 

However, my main question is duplication do we want to add a bit more rigor to the existing contributions.rst file (by just adding a best practice section under the git section) or so we want to keep them in both places? Maybe we provide a link from the contributors section to our wiki page. Please let me know what you guys think?

P.S: What is the intent of the separate LEARNING.md? Is it to capture individual learnings? How do we harmonize all of the similar work streams ?

**Comment by datapythonista** [comment]
You call, but just two things to consider.

I see the GitHub wiki just a fancy interface to a secondary github repo of markdowns (GitHub literally uses a git repo to store the files). Since everybody using this repo is able to work directly with the markdown files, and it probably adds value that the changes to the pages are done through a PR process, and not individual edits, I'd personally disable the wiki and keep things centralized in the main project repo.

Unrelated, the first actual task for pandas will be to write a better contributing documentation for the project. This will have a very big overlap with the contributing documentation you're writing. I'd move forward for now without worrying much about that, but probably worth to keep in mind when making any decision.

**Comment by sparalic** [comment]
@datapythonista I agree that it makes it easier for everyone to contribute ! I’ll keep mining those nuggets out of the PR messages. Thanks!

**Comment by martinagvilas** [comment]
Third attempt, but I think I got it right now. I couldn't find the commit that deleted the files, but I force-pushed everything into a clean state now. Let me know if its alright.

**Review by datapythonista — APPROVED** [review]
great job! those things are quite tricky, I'm also confused some times with some github messes, but they'll surely happen less often with time. :)
